extensions/smallrye-graphql/deployment/src/main/java/io/quarkus/smallrye/graphql/deployment/SmallRyeGraphQLProcessor.java

```diff
@@ -156,11 +155,7 @@ void buildExecutionService(
         recorder.createExecutionService(beanContainer.getValue(), schema);
 
         // Make sure the complex object from the application can work in native mode
-        for (String c : getClassesToRegisterForReflection(schema)) {
-            DotName name = DotName.createSimple(c);
-            org.jboss.jandex.Type type = org.jboss.jandex.Type.create(name, org.jboss.jandex.Type.Kind.CLASS);
-            reflectiveHierarchyProducer.produce(new ReflectiveHierarchyBuildItem(type, index));
-        }
+        reflectiveClassProducer.produce(new ReflectiveClassBuildItem(true, true, getSchemaJavaClasses(schema)));
```

Comment:
I'm not sure it's the right fix.

What I think would be right is:
- register the owner with a class build item
- register the return types of the GraphQL methods with a hierarchy build item

If you have a hierarchy of complex objects for the return type, I think what you did will not work.

Reply:
We already know the hierarchy as the model build that, but I am happy to change as you suggest. Let me add a test case with that scenario then we can see.

Reply:
Hey @gsmet - I added some tests to test parent and child classes. All seems ok. The model we build already discover all the classes needed. So this fix should be ok. Let me know.